src/provider/EventRecorder.ts

```diff
@@ -103,7 +112,7 @@ export class EventRecorder {
 				StorageUtil.clearEvents(eventsJson);
 			}
 			this.isFlushingEvents = false;
-			if (needsFlushTwice) {
+			if (result && needsFlushTwice) {
```

Comment:
What scenarios need to use `needsFlushTwice` ？

Reply:
By default, the maximum localStorage cache is 1024*1024, and the maximum payload of a request is 512*1024. When a request cannot send all events, we need to flush twice to send the remaining events.